scheduler-server/sql/scheduler/tables/scheduling_specification_goals.sql

```diff
@@ -1,12 +1,12 @@
 create table scheduling_specification_goals (
   specification_id integer not null,
   goal_id integer not null,
-  priority integer not null,
+  priority integer not null constraint non_negative_specification_goal_priority check (priority >= 0),
 
   constraint scheduling_specification_goals_primary_key
     primary key (specification_id, goal_id),
   constraint scheduling_specification_goals_unique_priorities
-    unique (specification_id, priority),
+    unique (specification_id, priority) deferrable initially deferred,
```

Comment:
This is a really neat solution to the priority uniqueness constraint problem :+1:

Reply:
I actually kinda didn't like it, because if something goes wrong you won't know until the end of the transaction. But I couldn't find any way to guarantee the order of triggered updates. (The internet said operating on an ordered view of the table would work, but that didn't seem to do it.)